server/routes/auth.ts

```diff
@@ -133,6 +147,120 @@ authRoutes.post('/login', async (req, res, next) => {
   }
 });
 
+authRoutes.post('/jellyfin', async (req, res, next) => {
+  const settings = getSettings();
+  const userRepository = getRepository(User);
+  const body = req.body as {
+    username?: string;
+    password?: string;
+    hostname?: string;
+  };
+
+  //Make sure jellyfin login is enabled, but only if jellyfin is not already configured
+  if (
+    settings.main.mediaServerType != MediaServerType.JELLYFIN &&
+    settings.jellyfin.hostname != ''
+  ) {
+    return res.status(500).json({ error: 'Jellyfin login is disabled' });
+  } else if (!body.username || !body.password) {
+    return res
+      .status(500)
+      .json({ error: 'You must provide an username and a password' });
+  } else if (settings.jellyfin.hostname != '' && body.hostname) {
+    return res
+      .status(500)
+      .json({ error: 'Jellyfin hostname already configured' });
+  } else if (settings.jellyfin.hostname == '' && !body.hostname) {
+    return res.status(500).json({ error: 'No hostname provided.' });
+  }
+
+  try {
+    const hostname =
+      settings.jellyfin.hostname != ''
+        ? settings.jellyfin.hostname
+        : body.hostname;
+    // First we need to attempt to log the user in to jellyfin
+    const jellyfinserver = new JellyfinAPI(hostname ?? '');
+    settings.jellyfin.name = await jellyfinserver.getServerName();
+
+    const account = await jellyfinserver.login(body.username, body.password);
+
+    // Next let's see if the user already exists
+    let user = await userRepository.findOne({
+      where: { jellyfinId: account.User.Id },
+    });
+
+    if (user) {
+      // Let's check if their authtoken is up to date
+      if (user.jellyfinAuthToken !== account.AccessToken) {
+        user.jellyfinAuthToken = account.AccessToken;
+      }
+
+      // Update the users avatar with their jellyfin profile pic (incase it changed)
+      if (typeof account.User.PrimaryImageTag !== undefined) {
+        user.avatar = `${hostname}/Users/${account.User.Id}/Images/Primary/?tag=${account.User.PrimaryImageTag}&quality=90`;
+      } else {
+        user.avatar = '/images/os_logo_square.png';
+      }
+      user.email = account.User.Name;
+      user.jellyfinUsername = account.User.Name;
+
+      if (user.username === account.User.Name) {
+        user.username = '';
+      }
+      await userRepository.save(user);
+    } else {
+      // Here we check if it's the first user. If it is, we create the user with no check
+      // and give them admin permissions
+      const totalUsers = await userRepository.count();
+
+      if (totalUsers === 0) {
+        user = new User({
+          email: account.User.Name,
+          jellyfinUsername: account.User.Name,
+          jellyfinId: account.User.Id,
+          jellyfinAuthToken: account.AccessToken,
+          permissions: Permission.ADMIN,
+          avatar:
+            typeof account.User.PrimaryImageTag !== undefined
```

Comment:
```suggestion
            account.User.PrimaryImageTag
```

`undefined` is falsy, so you don't need to do `typeof` here.